lib/OpenQA/Utils.pm

```diff
@@ -958,4 +961,18 @@ sub regex_match ($regex_string, $string) {
     return $match;
 }
 
+# Returns a microsecond value suitable for use with "usleep", for the first iteration it returns the minimum value plus
+# random 0-1 second padding, and starting with the second iteration the delay increases by one second plus 0-1 second
+# padding, up to the maximum
+sub usleep_backoff ($iteration, $min_seconds, $max_seconds, $padding = int(rand(ONE_SECOND_IN_MICROSECONDS))) {
+
+    # To allow for backoff to be disabled easily
+    return 0 if $min_seconds == 0;
+
+    my $delay
+      = ($min_seconds * ONE_SECOND_IN_MICROSECONDS) + (ONE_SECOND_IN_MICROSECONDS * ($iteration - 1)) + $padding;
+    my $max = $max_seconds * ONE_SECOND_IN_MICROSECONDS;
+    return $delay > $max ? $max : $delay;
```

Comment:
One could `use List::Util qw(min);` here:
```suggestion
    return min($max, $delay);
```

Reply:
I tend to avoid `List::Util` when it doesn't make a big difference for readability, because in many cases it can turn into a performance regression (often because of fake code blocks). For `max` and `min` i'm actually not sure how well they scale.

Reply:
I guess it's a little cleaner if i merge it with the previous line too. 🤔

Reply:
Regarding performance, I think in this case we don't need to care because this code isn't called very often.
Also https://metacpan.org/pod/List::Util::XS says that it will use the XS version always.

Reply:
I would have expected that `List::Util` would be faster as it is likely implemented natively (so there's more native code executed and less Perl code). Seems like this is at least true at this point. I guess it really doesn't matter here, though.

Reply:
Despite it being implemented in XS there are performance bottlenecks. For example `reduce {...} 1..10` is problematic because the `{...}` blocks here are not real blocks, but functions made to look like blocks. These have a lot more overhead than real blocks that truly native Perl features like `grep {...} 1..10` can use. This is similar to how `for` loops are preferable to iterator APIs like `$collection->for_each(sub ($item) {...})` because loops are so much more efficient.